src/main/groovy/com/bmuschko/gradle/clover/InstrumentCodeAction.groovy

```diff
@@ -95,7 +95,13 @@ class InstrumentCodeAction implements Action<Task> {
                 }
 
                 getMethodContexts().each {
-                    ant.methodContext(name: it.name, regexp: it.regexp)
+                    ant.methodContext(
```

Comment:
If statement and method contexts have the same fields then you can just add them to `CloverContextConvention`. Otherwise, we'll need to have two different classes. In terms of the class hierarchy it depends on what they share. If they share `name` and `regexp` then you can probably use `CloverContextConvention` as super class for `CloverMethodContextConvention`.